client/src/components/Login.js

```diff
@@ -61,18 +65,57 @@ function Login(props) {
   const classes = useStyles();
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
+  const [openSnack, setOpenSnack] = useState(false);
+  const [snackText, setSnackText] = useState("");
   const value = useContext(UserContext);
+  const vertical = "top";
+  const horizontal = "center";
+
+  const handleSnack = (props) => {
+    setSnackText(props);
+    setOpenSnack(true);
+  };
+
+  const handleCloseSnack = (event, reason) => {
+    if (reason === "clickaway") {
+      return;
+    }
+
+    setOpenSnack(false);
+  };
 
-  const handleSubmit = async (e) => {
+  const handleLogin = async (e) => {
     e.preventDefault();
-    const loginSuccess = value.handleLogin(email, password);
-    if (loginSuccess) props.history.push("/dashboard");
+    fetch("/auth/login", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        email: email,
+        password: password,
+      }),
+    })
+      .then((response) => response.json())
+      .then(function (response) {
+        if (response.status === "success") {
+          console.log("Success:", email);
+          const loginSuccess = value.handleLogin(email, password);
+          if (loginSuccess) props.history.push("/dashboard");
+        } else {
+          handleSnack(response.message);
+          console.log(response.message);
+        }
```

Comment:
It seems that you have moved handle login from context to login component which makes context useless, any reason why the handle login have been moved to login component?

Reply:
I couldn't figure out how to get the response.message to handleSnack when using context. If there is a way to do that, I'm all for it.

Reply:
Rather than return a boolean from handleLogin in context, we could return the response object, then check its message here

Reply:
I tried doing that before, the problem is that the object returned is a promise, and that promise doesn't work well with if/else statements for whatever reason. For instance, I tried expanding on the original if statement (which did work with the promise) by adding an else that used handleSnack(loginSuccess.message);, but it didn't work. I'm definitely open to using the UserContext, but I spent a fair bit of time trying to make it work with that and I couldn't figure it out. If someone has more experience handling promises, that'd be helpful!

Reply:
hmm. Yeah, promises can be tricky. Were you approaching it like this?
```
const handleLogin = async (e) => {
  const response = await value.handleLogin(email, password);
  if (response.status === "success") //do stuff
  else handleSnack(response.message);
```
if that doesn't work, why don't we make the API call here and then call handleLogin from Context just to set the user object to true - that's the important part of using context, not the actual location of the API calls

Reply:
That worked, thank you so much! I think I didn't put await in initially, and that was causing the problems.

Reply:
I'm going to try to put in something similar in SignUp.js so the code isn't repeated between UserContext and SignUp.